### ISSUE TITLE
A way to createModuleEx with config fixities?

### ISSUE THREAD

**Issue by eddiemundo** [ndmitchell/hlint#1324]
In HLSs hlint plugin, if the flag `HLINT_ON_GHC_LIB` is not defined then it uses `createModuleEx` to create a `ModuleEx` but this ignores fixities found in the `.hlint.yaml` ([#2418](https://github.com/haskell/haskell-language-server/issues/2418#issue-1066910078)). I figure adding a function like
```haskell
createModuleWithFixities :: [(String, Fixity)] -> ApiAnns -> Located HsModule -> ModuleEx
createModuleWithFixities fixities anns ast = 
  ModuleEx (applyFixities (fixitiesFromModule ast ++ fixities) ast) anns
```
would allow us to use `ApiAnns` from the ghc api, and also use fixities from the `.hlint.yaml` file, at the small expense of duplicating the `toFixity` function to convert `ParseFlags` to `[(String, Fixity)]`.

Alternatively we could have `createModuleWithParseFlags` take a `ParseFlags` and use the not exposed `ghcFixitiesFromParseFlags` to do the work, and have no duplication of code on the client side.

Or we could also expose `ghcFixitiesFromParseFlags` or `Fixity.toFixity` but meh.

**Comment by ndmitchell** [comment]
Your choice to add the single function seems not harmful, and lets us avoid making decisions for what gets exposed, so seems like  a good idea.
